scripts/benchmark_all.sh

```diff
@@ -74,17 +84,14 @@ do
   for benchmark in $benchmarks
   do
     echo "Running $benchmark for $commit... (single-threaded)"
-    ( "${build_folder}"/"$benchmark" -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st.log"
+    ( "${build_folder}"/"$benchmark" -r ${runs} -w ${warmup_seconds} -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st.log"
 
     if [ "$benchmark" = "hyriseBenchmarkTPCH" ]; then
       echo "Running $benchmark for $commit... (single-threaded, SF 0.01)"
-      ( "${build_folder}"/"$benchmark" -s .01 -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st_s01.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st_s01.log"
+      ( "${build_folder}"/"$benchmark" -s .01 -r ${runs} -w ${warmup_seconds} -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st_s01.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st_s01.log"
 
-      echo "Running $benchmark for $commit... (single-threaded, SF 1.0)"
-      ( "${build_folder}"/"$benchmark" -s 1 -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st_s1.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_st_s1.log"
-
-      echo "Running $benchmark for $commit... (multi-threaded, clustered for pruning)"
-      ( "${build_folder}"/"$benchmark" --scheduler --clients ${num_mt_clients} -s 10 --dont_cache_binary_tables --clustering=Pruning -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt_clustered.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt_clustered.log"
+      echo "Running $benchmark for $commit... (multi-threaded, shuffled)"
+      ( "${build_folder}"/"$benchmark" --scheduler --clients ${num_mt_clients} -s 10 -m Shuffled --dont_cache_binary_tables -o "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt_clustered.json" 2>&1 ) | tee "${build_folder}/benchmark_all_results/${benchmark}_${commit}_mt_clustered.log"
```

Comment:
No warm up here?

Reply:
Does the warmup make sense for shuffled runs?

Reply:
To be honest, I don't know. I am not even sure why a warmup makes sense in the other cases. Most queries are trivial to optimize and the data is in DRAM anyways.

Do you have a clue what's making a difference here?

Reply:
No, I had a quick look at it. The difference between first and second query run seems to be largest for the first benchmark items, 20% and more for TPC-H SF 10. I believe I have seen more for the JOB, maybe due to optimization time. For the TPC-H, I verified that the optimization time was not the issue so the CPU cache is the only thing that would make sense to me.